pulsar/test/suite.go

```diff
@@ -56,41 +68,93 @@ func (suite *PulsarTestSuite) SetupSuite() {
 	suite.shutdownFunc = func() {
 		defer func() {
 			suite.Client.Close()
+			formmatedScript := fmt.Sprintf(pulsarStopCommand, randomContainerName)
+			outbytes, err := exec.Command("/bin/sh", "-c", formmatedScript).CombinedOutput()
+			if err != nil {
+				suite.FailNow("failed to stop pulsar", err.Error(), string(outbytes))
+			}
 		}()
 	}
 }
-
+func (suite *PulsarTestSuite) checkPulsarIsAlive() bool {
+	kaURL := fmt.Sprintf(pulsarKAURL, suite.DefaultTestConfig.AdminUrl)
+	req, err := http.NewRequest(http.MethodGet, kaURL, nil)
+	if err != nil {
+		suite.FailNow("failed to create request", err.Error())
+	}
+	client := http.Client{Timeout: 5 * time.Second}
+	resp, err := client.Do(req)
+	if err == nil && resp.StatusCode == http.StatusOK {
+		suite.T().Log("pulsar started")
+		resp.Body.Close()
+		return true
+	}
+	return false
+}
 func (suite *PulsarTestSuite) TearDownSuite() {
 	suite.T().Log("tear down suite")
 	suite.shutdownFunc()
-	exec.Command("/bin/sh", "-c", pulsarStopCommand).Run()
+	killPortProcess(suite.AppPortStart)
+	killPortProcess(suite.AdminPortStart)
 }
 
 func (suite *PulsarTestSuite) SetupTest() {
 	suite.T().Log("setup test")
 }
 
-func (suite *PulsarTestSuite) startPulsar() {
+func (suite *PulsarTestSuite) TearDownTest() {
+	suite.T().Log("tear down test")
+	// clear all pulsar topics messages
+	if err := suite.clearAllMessages(); err != nil {
```

Comment:
Another approach is to fail if there are messages and force each test to cleanup 
This can also cover unexpected additional messages